src/mono/wasm/runtime/exports.ts

```diff
@@ -161,16 +170,43 @@ function initializeImportsAndExports(
     // this could be overriden on Module
     if (!module.onRuntimeInitialized) {
         module.onRuntimeInitialized = mono_wasm_on_runtime_initialized;
+        module.ready = module.ready.then(() => {
+            return mono_wasm_runtime_is_initialized;
```

Comment:
Are we really returning here a function?
Maybe I don't understand how `module.ready` is awaited..

Reply:
`mono_wasm_runtime_is_initialized` is a promise
`mono_wasm_on_runtime_initialized` is a method which is resolving it